src/dom.js

```diff
@@ -35,44 +33,40 @@ export class Js {
     this.log = new Log(enableLogging);
   }
 
-  injectWithText(text, url) {
-    return new Promise(resolve => {
-      let script = document.createElement('script');
+  prepareWithText(text, url) {
+    let script = document.createElement('script');
 
-      this.log.info(`Creating <script />-tag with text for ${url}.`);
+    this.log.info(`Creating <script />-tag with text for ${url}.`);
 
-      script.defer = false;
-      script.async = false;
+    script.defer = false;
+    script.async = false;
 
-      script.setAttribute('data-dactylographsy-url', url);
+    script.setAttribute('data-dactylographsy-url', url);
 
-      script.text = `
-        ${text}
-        //# sourceURL=${url}
-      `;
+    script.text = `
+      ${text}
+      //# sourceURL=${url}
+    `;
 
-      if (this.injectInto) {
-        this.log.info(`Injecting <script />-tag with url: ${url}.`);
-
-        resolve(this.injectInto.appendChild(script));
-      } else { resolve(script); }
-    });
+    return Promise.resolve(script);
   }
 
-  injectWithUrl(urls, whichUrl = 'printed') {
-    return new Promise(resolve => {
-      // Create script element and set its type
-      let
-        script = document.createElement('script'),
-        url = urls[whichUrl];
+  prepareWithUrl(urls, whichUrl = 'printed') {
+    const
+      urlKeys = Object.keys(urls),
+      scriptTags = {};
+
+    urlKeys.forEach((urlKey) => {
```

Comment:
I think we should filter the urlKeys here, because otherwise urls for `hash` and `singularBy` are also created, we are already in `prepareWithUrls` so filtering for `printed` and `raw` should be fine